### DISCUSSION TITLE
sqlalchemy 1.4.8 conflic

### DISCUSSION THREAD

**Discussion by haidv1992** [fastapi-users/fastapi-users#1215] [Q&A]
when i use ID instead of uuid 
it warring not exited mapped_column

class User(SQLAlchemyBaseUserTableUUID, Base):
    id: Mapped[int] = mapped_column(Integer, primary_key=True)

when  use uuid then use alembic to migrate so error because not auto import function to migration

but when i use

class User(SQLAlchemyBaseUserTableUUID, Base):
    id : Column(Integer, primary_key=True)
    
 after logged then get user/me i got status 401 , i think this error because  it not define id

**Comment by frankie567** [comment]
If you want to override the type of ID, you need to inherit from `SQLAlchemyBaseUserTable`, not from `SQLAlchemyBaseUserTableUUID`:

```py
class User(SQLAlchemyBaseUserTable[int], Base):
    id: Mapped[int] = mapped_column(Integer, primary_key=True)
```

Ref: https://fastapi-users.github.io/fastapi-users/11.0/configuration/databases/sqlalchemy/#create-the-user-model

**Reply by arman-async** [reply]
Thank you so much! I had guessed from reading the source code that I should do this, but you made me try it quickly instead of spending too much time searching. I really appreciate it.
